extra_data/tests/test_reader_mockdata.py

```diff
@@ -511,7 +511,7 @@ def test_select(mock_fxe_raw_run):
         sel.select('FXE_DET_LPD1M-1/DET/0CH0:xtdf', '*') \
            .keys_for_source('FXE_DET_LPD1M-1/DET/0CH0:xtdf')
     assert sel.keys_for_source('FXE_DET_LPD1M-1/DET/0CH0:xtdf') == \
-        sel.select({'FXE_DET_LPD1M-1/DET/0CH0:xtdf': {}}) \
+        sel.select({'FXE_DET_LPD1M-1/DET/0CH0:xtdf': set()}) \
```

Comment:
The previous implementation allowed an empty dict in place of an empty set, but now that's an error. I hope no-one was using that, but we could allow it if necessary.

Reply:
OK, done. I haven't added a warning, just let it carry on working.